Display more user-friendly error messages

NorbiPeti opened the pull request:
Instead of displaying the Java exception, this change displays a more user-friendly message for some common network-related issues.

Fixes mastodon/mastodon-android#667

Please let me know if this is a good way to implement this or what would be the best way to do it, if not.

samhenrigold commented:
Hey there, I’m the lead product designer on the first party Mastodon apps. Thanks for submitting this PR! Agreed that we should do better with error messaging. Would you be okay with some copy modifications? Mostly to keep things consistent:
```xml
<string name="could_not_reach_server">Couldn’t reach the server. Check your connection and try again?</string>
<string name="connection_timed_out">The request timed out. Check your connection and try again?</string>
<string name="server_error">Something went wrong talking with your server. It’s probably not your fault. Try again?</string>
<string name="not_found"It could’ve been deleted, or maybe it never existed at all.</string>
```

I’d cut the unknown error part since we can just return whatever error the server gives. “Unknown” implies to me that we’re too deep into abstractions to know what the error was.

NorbiPeti commented:
Thanks for the feedback! I've updated the strings and removed the unknown error string and instead display the original message directly. My thought process with that was to indicate if there's no proper message for an error yet but the end user doesn't really have to know about that (it doesn't help much).
